### PR TITLE
Fix assembly on ARM32

### PR DESCRIPTION
**Pull request by kant2002**
When I copy over that file, I do not realize that these 2 functions do not compile properly

So far left following
- Produce libObjWriter.so for ARM
- Rewrite RhGetRuntimeHelperForType in C# as per this discussion https://github.com/dotnet/runtimelab/pull/1294#issuecomment-874430938

### CODE REVIEW

**Comment by jkotas** [comment]
GcProbe.asm is for Windows. GcProbe.S is for non-Windows.

**Comment by kant2002** [comment]
So far, outstanding items assuming that I write assembly correctly.
- `libobjwriter` built just fine using `./build.sh nativeaot.objwriter`. That's probably expected.
- Rewrite RhGetRuntimeHelperForType in C# (required for completeness, not for hacking)
- I built just `nativeaot` subset, other blocked. See below.
- I now stuck on Nuget issue, since I cannot build `libs` subset on Raspberry, [due to](https://github.com/NuGet/Home/issues/10968) memory limitations. I catch command line which execute restore and request 1.2Gb in total, so maybe I can find a way how to reduce memory pressure. 
- Once I build `libs` I believe I can run `nativeaot.packages` just fine. Building that subset blocked on `libs`.

File: src/coreclr/nativeaot/Runtime/arm/GcProbe.S [hunk 1]
```diff
@@ -7,14 +7,23 @@
     .global RhpGcPoll2
 
     LEAF_ENTRY RhpGcPoll
-        PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, 0
-        cbnz    w0, RhpGcPollRare // TrapThreadsFlags_None = 0
-        ret
+        // @todo: I'm assuming it's not OK to trash any register here. If that's not true we can optimize the
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/arm/GcProbe.S]: I do not think that this assumption is correct. It should be fine to trash any volatile registers here.

Comment (kant2002) [reply src/coreclr/nativeaot/Runtime/arm/GcProbe.S]: Should I fix that in windows asm too?

File: src/coreclr/nativeaot/Runtime/arm/GcProbe.S [hunk 2]
```diff
@@ -7,14 +7,16 @@
     .global RhpGcPoll2
 
     LEAF_ENTRY RhpGcPoll
-        PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, 0
```
Comment (kant2002) [inline src/coreclr/nativeaot/Runtime/arm/GcProbe.S:10]: Technically I can create similar macro for ARM

File: src/coreclr/nativeaot/Runtime/arm/GcProbe.S (hunk 1 above)
Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/arm/GcProbe.S]: Yes, it would be nice.

File: src/coreclr/nativeaot/Runtime/arm/GcProbe.S (hunk 2 above)
Comment (jkotas) [reply src/coreclr/nativeaot/Runtime/arm/GcProbe.S:10]: I do not think that it is necessary.

**Comment by kant2002** [comment]
Thanks

**Pull request merged**